aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -322,7 +322,7 @@ impl AptosVM {
     pub fn failed_transaction_cleanup(
         &self,
         error_code: VMStatus,
-        gas_meter: &impl AptosGasMeter,
+        gas_meter: &mut impl AptosGasMeter,
```

Comment:
Seems like code coverage annotations are useful after all, we do not test this path?

Reply:
No, they seem very wrong. This is tested in nearly every test I added. Otherwise they wouldn't be able to mutate the gas meter, which is verified in the test. Other cases are similarly wrong.

Reply:
If they are wrong, do we need this tool? cc @sausagee